File: 1512-number-of-good-pairs.cpp
/*
1512. Number of Good Pairs

Submitted: October 18, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 8.88 MB (beats 92.44%)
*/

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int res = 0;
        for (int i = 0; i < nums.size(); ++i) {
            for (int j = i + 1; j < nums.size(); ++j) {
                if (nums[i] == nums[j]) {
                    res++;
                }
            }
        }
        return res;
    }
};

File: 1512-number-of-good-pairs.py
"""
1512. Number of Good Pairs

Submitted: November 20, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 17.92 MB (beats 44.64%)
"""

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        c = Counter(nums)
        res = 0
        for k, v in c.items():
            res += v * (v - 1) // 2
        return res